### PR TITLE
Fix owner_alias to owner migration by replacing GetOK() with Get()

### PR DESCRIPTION
**Pull request by taimoor-at-opslevel**
## Issues

https://gitlab.com/jklabsinc/OpsLevel/-/issues/9951

TL;DR customer reported a bug where changing the `owner` field for an `owner_alias` and vice versa causes an error to be shown. That error is caused by the if statement that relies on `GetOk()`. 

## Changelog

- [x] Replace if statements
- [x] Make a `changie` entry

## Tophatting

All of these pass and there are no errors shown, only deprecation warnings when owner_alias is used.

Create a new service with owner_alias.

```
  # opslevel_service.bugsv2 will be created
  + resource "opslevel_service" "bugsv2" {
      + id           = (known after apply)
      + last_updated = (known after apply)
      + name         = "Bugged Service 2"
      + owner_alias  = "bugging_team"
    }

Plan: 1 to add, 0 to change, 0 to destroy.
```

Replace the owner_alias with owner.

```
  # opslevel_service.bugsv2 will be updated in-place
  ~ resource "opslevel_service" "bugsv2" {
        id          = "Z2lkOi8vb3BzbGV2ZWwvU2VydmljZS84NzUwOQ"
        name        = "Bugged Service 2"
      + owner       = "bugging_team"
      - owner_alias = "bugging_team" -> null
        tags        = []
        # (1 unchanged attribute hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

Opposite way works too.

```
  # opslevel_service.bugsv2 will be updated in-place
  ~ resource "opslevel_service" "bugsv2" {
        id           = "Z2lkOi8vb3BzbGV2ZWwvU2VydmljZS84NzUwOQ"
        name         = "Bugged Service 2"
      - owner        = "bugging_team" -> null
      + owner_alias  = "bugging_team"
        tags         = []
        # (2 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

Tested a completely different alias just out of curiosity.

```
  # opslevel_service.bugsv2 will be updated in-place
  ~ resource "opslevel_service" "bugsv2" {
        id           = "Z2lkOi8vb3BzbGV2ZWwvU2VydmljZS84NzUwOQ"
        name         = "Bugged Service 2"
      ~ owner_alias  = "bugging_team" -> "platform"
        tags         = []
        # (2 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

Updating to use both owner and owner_alias should fail.

```
  # opslevel_service.bugsv2 will be updated in-place
  ~ resource "opslevel_service" "bugsv2" {
        id           = "Z2lkOi8vb3BzbGV2ZWwvU2VydmljZS84NzUwOQ"
        name         = "Bugged Service 2"
      + owner        = "Platform"
        tags         = []
        # (3 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
│ Error: can pass only one of: 'owner' or 'owner_alias'
│ 
│   with opslevel_service.bugsv2,
│   on main.tf line 13, in resource "opslevel_service" "bugsv2":
│   13: resource "opslevel_service" "bugsv2" {
│ 
╵
task: Failed to run task "apply": task: Failed to run task "terraform-command": exit status 1
```

So should creating a new service with both fields.

```
  # opslevel_service.bugsv3 will be created
  + resource "opslevel_service" "bugsv3" {
      + id           = (known after apply)
      + last_updated = (known after apply)
      + name         = "Bugged Service 3"
      + owner        = "platform"
      + owner_alias  = "platform"
    }

Plan: 1 to add, 1 to change, 0 to destroy.

│ Error: can pass only one of: 'owner' or 'owner_alias'
│ 
│   with opslevel_service.bugsv3,
│   on main.tf line 18, in resource "opslevel_service" "bugsv3":
│   18: resource "opslevel_service" "bugsv3" {
│ 
╵
task: Failed to run task "apply": task: Failed to run task "terraform-command": exit status 1
```

### CODE REVIEW

File: opslevel/resource_opslevel_service.go [hunk 1]
```diff
@@ -188,13 +188,13 @@ func reconcileTags(d *schema.ResourceData, service *opslevel.Service, client *op
 func resourceServiceCreate(d *schema.ResourceData, client *opslevel.Client) error {
 	// owner_alias is deprecated, allow user to use one or the other but not both.
 	var ownerField *opslevel.IdentifierInput
-	owner, ownerUsed := d.GetOk("owner")
-	ownerAlias, ownerAliasUsed := d.GetOk("owner_alias")
-	if ownerUsed && ownerAliasUsed {
+	owner := d.Get("owner")
```
Comment (taimoor-at-opslevel) [inline opslevel/resource_opslevel_service.go:191]: I replaced `GetOk()` with `Get()`. Reason being

>GetOk returns the data for the given key and whether or not the key **has been set to a non-zero value at some point.**

>The first result will not necessarilly be nil if the value doesn't exist. **The second result should be checked to determine this information.**

GetOk() isn't useful here, we still need to check the value. We need to see if the value is currently set, not if it was ever set before. 

---

Get() is the better function to use, because

>If the key does exist in the schema but doesn't exist in the configuration, then the default value for that type will be returned. For strings, this is "", for numbers it is 0, etc.

---

Source: https://pkg.go.dev/github.com/hashicorp/terraform/helper/schema#ResourceData.Get

Comment (rocktavious) [reply opslevel/resource_opslevel_service.go:191]: 👍  - this might be a logic problem across the codebase then - lets make a ticket to audit this.

**Review by rocktavious — APPROVED** [review]

File: opslevel/resource_opslevel_service.go (hunk 1 above)
Comment (taimoor-at-opslevel) [reply opslevel/resource_opslevel_service.go:191]: Added new ticket to backlog: https://github.com/OpsLevel/team-platform/issues/152
